index.js

```diff
@@ -68,56 +79,66 @@ class Geocoder extends PureComponent {
     this.select.focus()
   }
 
-  loadOptions = (input) => {
-    const {apiKey, boundary, focusPoint, search} = this.props
-    return search({
-      apiKey,
-      boundary,
-      focusPoint,
-      text: input
-    }).then((geojson) => {
-      const options = geojson && geojson.features
-        ? geojson.features.map(this.featureToOption)
-        : []
-      // insert geolocate option if geolocate is enabled
-      if (this.props.geolocate && 'geolocation' in navigator && input === '') {
-        // TODO: handle option rendering in renderGeolocateOption with <Icon type={option.icon} />
-        options.push({
-          icon: 'location-arrow',
-          label: this.props.useLocationText,
-          value: 'TBD',
-          geolocate: true,
-          feature: {
-            properties: {label: 'test'},
-            geometry: {
-              coordinates: [0, 0]
-            }
-          }
+  loadOptions = (input, callback) => {
+    const {apiKey, boundary, focusPoint, geolocate, search} = this.props
+    if (!input) {
+      if (geolocate && 'geolocation' in navigator) {
+        callback(null, {
+          options: this.defaultOptions()
         })
-      } else {
-        /* geolocation IS NOT available, do nothing */
       }
-      this.cacheOptions(options)
-      return {options}
-    })
+    } else {
+      search({
+        apiKey,
+        boundary,
+        focusPoint,
+        text: input
+      }).then((geojson) => {
+        const options = geojson && geojson.features
+          ? geojson.features.map(this.featureToOption)
+          : []
+        this.cacheOptions(options)
+        callback(null, {options})
+      }).catch((error) => {
+        callback(error)
+      })
+    }
   }
 
   _onChange = (value) => {
-    if (value && value.geolocate) {
-      value.label = this.props.findingLocationText
-      this.setState({value})
-      navigator.geolocation.getCurrentPosition((position) => {
-        const result = {
-          icon: 'location-arrow',
-          label: `My location (${position.coords.longitude.toFixed(5)}, ${position.coords.latitude.toFixed(5)})`,
-          value: `${position.coords.longitude},${position.coords.latitude}`,
-          geolocate: true
+    const {apiKey, findingLocationText, onChange, reverseSearch} = this.props
+    if (value && value.value === GEOLOCATE_VALUE) {
+      this.setState({
+        ...this.state,
+        value: {
+          label: findingLocationText
         }
-        this.setState({value: result})
-        this.props.onChange && this.props.onChange(value && this.options[value.value])
+      })
+      window.navigator.geolocation.getCurrentPosition((position) => {
+        reverseSearch({
+          apiKey,
+          point: position.coords
+        }).then((geojson) => {
+          const value = this.featureToOption(geojson.features[0])
```

Comment:
@trevorgerhardt what if there's an error in the `reverseSearch`?  Wouldn't we want to just return the `position.coords` as a fallback?